lib/handler/configurator/proxy.c

```diff
@@ -99,6 +100,10 @@ static void update_ssl_ctx(SSL_CTX **ctx, X509_STORE *cert_store, int verify_mod
     CRYPTO_add(&cert_store->references, 1, CRYPTO_LOCK_X509_STORE);
     if (verify_mode == -1)
         verify_mode = (*ctx)->verify_mode;
+    if (session_cache == NULL && delete_session_cache == 0) {
+        session_cache = h2o_socket_ssl_get_session_cache(*ctx);
+        h2o_socket_ssl_set_session_cache(*ctx, NULL); /* prevent the cache from being disposed */
+    }
```

Comment:
Sorry, I do not get this.

My understanding is that it is a requirement to have a session cache associated for every SSL_CTX that has been set up with both capacity and lifetime set to a non-zero value.

However, in this _if_ block, you seem to be detaching an session cache from an existing SSL_CTX.

Doesn't this cause a problem?

As an example, consider the following configuration. Would upstream connections for a.example.com use the session cache?

```yaml
proxy.ssl-session-cache: ON
hosts:
  a.example.com:
    /:
      proxy.reverse.url: ...
  b.example.com:
    /:
      proxy.reverse.url: ...
    proxy.ssl.session-cache: OFF
```